kbcstorage/buckets.py

```diff
@@ -0,0 +1,164 @@
+        Returns:
+            response_body: The parsed json from the HTTP response.
+
+        Raises:
+            requests.HTTPError: If the API request fails.
+        """
+        headers = {'X-StorageApi-Token': self.token}
+
+        url = '{}/{}/tables'.format(self.base_url, bucket_id)
+        params = {}
+        if include is not None and isinstance(include, list):
+            params['include'] = ','.join(include)
+        return self.get(url, headers=headers, params=params)
+
+    def detail(self, bucket_id):
+        """
+        Retrieves information about a given bucket.
+
+        Args:
+            bucket_id (str): The id of the bucket.
+
+        Raises:
+            requests.HTTPError: If the API request fails.
+        """
+        url = '{}/{}'.format(self.base_url, bucket_id)
+        headers = {'X-StorageApi-Token': self.token}
+
+        return self.get(url, headers=headers)
+
+    def create(self, name, stage='in', description='', backend=None):
+        """
+        Create a new bucket.
+
+        Args:
+            name (str): The new bucket name (only alphanumeric and underscores)
+            stage (str): The new bucket stage. Can be one of ``in`` or ``out``.
+                Default ``in``.
+            description (str): The new bucket description.
+            backend (str): The new bucket backend. Cand be one of
+                ``snowflake``, ``redshift`` or ``mysql``. Default determined by
+                project settings.
+        Returns:
+            response_body: The parsed json from the HTTP response.
+
+        Raises:
+            requests.HTTPError: If the API request fails.
+        """
+        # Separating create and link into two distinct functions...
+        headers = {
+            'X-StorageApi-Token': self.token,
+            'Content-Type': 'application/x-www-form-urlencoded'
+        }
+        # Need to check args...
+        body = {
+            'name': name,
+            'stage': stage,
+            'description': description,
+            'backend': backend
+        }
+
+        return self.post(self.base_url, headers=headers, data=body)
+
+    def delete(self, bucket_id, force=False):
+        """
+        Delete a bucket referenced by ``bucket_id``.
+
+        By default, only empty buckets without dependencies (aliases etc) can
+        be deleted. The optional ``force`` parameter allows for the deletion
+        of non-empty buckets.
+
+        Args:
+            bucket_id (str): The id of the bucket to be deleted.
+            force (bool): If ``True``, deletes the bucket even if it is not
+                empty. Default ``False``.
+        """
+        # How does the API handle it when force == False and the bucket is non-
+        # empty?
+        url = '{}/{}'.format(self.base_url, bucket_id)
+        headers = {'X-StorageApi-Token': self.token}
+        params = {'force': force}
+        super().delete(url, headers=headers, params=params)
```

Comment:
I have a PR prepared which hides the base HTTP `Endpoint.get()` methods with and _underscore `Endpoint._get()` to avoid these namespace collisions.

Will push it soon

Reply:
ok, let's leave this for the other PR